[BUG]: Compilation error with nginx 1.26.3  nginx-datadog 1.6.1

### Module Version(s)

1.6.1

### Bug Report

After i posted this [issue](https://github.com/DataDog/nginx-datadog/issues/196) and it was resolved with this commit `629830997934c2fded64271a30d1fcc5ee174bec` 
now i bumped nginx-datadog to 1.6.1 which includes the fix commit but now i have this error 
```
In file included from /tmp/nginx-datadog/src/common/directives.cpp:1:
/tmp/nginx-datadog/src/common/directives.h:17:24: error: 'datadog::common::ngx_conf_post_file_exists' defined but not used [-Werror=unused-variable]
   17 | static ngx_conf_post_t ngx_conf_post_file_exists = {check_file_exists};
      |                        ^~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
gmake[3]: *** [CMakeFiles/ngx_http_datadog_objs.dir/build.make:90: CMakeFiles/ngx_http_datadog_objs.dir/src/common/directives.cpp.o] Error 1
gmake[3]: *** Waiting for unfinished jobs....
gmake[3]: Leaving directory '/tmp/nginx-datadog/.build'
gmake[2]: Leaving directory '/tmp/nginx-datadog/.build'
gmake[2]: *** [CMakeFiles/Makefile2:1350: CMakeFiles/ngx_http_datadog_objs.dir/all] Error 2
gmake[1]: *** [Makefile:146: all] Error 2
gmake[1]: Leaving directory '/tmp/nginx-datadog/.build'

make: *** [Makefile:24: build] Error 2
```

i tried to follow up and debug to find which commit introduces the problem and it this one `45ce45e2cc0ea63a88b204a2d2155983bc7df679`
introduced to this [PR](https://github.com/DataDog/nginx-datadog/pull/197)

nothing changed from our end. everything is still the same as declared in the first issue, i figured this might be a small fix, if not i can provide you will all the details you need  

### Reproduction Code

_No response_

### Error Logs

_No response_

### Operating System

_No response_

--- a/src/common/directives.cpp
+++ b/src/common/directives.cpp
@@ -7,6 +7,7 @@
 
 namespace datadog::common {
 
+#ifdef WITH_WAF
 char *check_file_exists(ngx_conf_t *cf, void *post, void *data) {
   assert(data != nullptr);
   ngx_str_t *s = (ngx_str_t *)data;
@@ -16,5 +17,6 @@ char *check_file_exists(ngx_conf_t *cf, void *post, void *data) {
   }
   return NGX_CONF_OK;
 }
+#endif
 
 }  // namespace datadog::common

--- a/src/common/directives.h
+++ b/src/common/directives.h
@@ -13,7 +13,9 @@ namespace datadog::common {
 /// directive actually exists on the filesystem.
 char *check_file_exists(ngx_conf_t *cf, void *post, void *data);
 
+#ifdef WITH_WAF
 /// Post handler for checking a filepath exists.
 static ngx_conf_post_t ngx_conf_post_file_exists = {check_file_exists};
+#endif
 
 }  // namespace datadog::common

--- a/src/datadog_directive.h
+++ b/src/datadog_directive.h
@@ -7,6 +7,7 @@ extern "C" {
 #include <ngx_http.h>
 }
 
+#include <array>
 #include <string_view>
 
 #include "string_util.h"

--- a/src/ngx_http_datadog_module.cpp
+++ b/src/ngx_http_datadog_module.cpp
@@ -316,7 +316,7 @@ static ngx_int_t datadog_module_init(ngx_conf_t *cf) noexcept {
   // Add default span tags.
   const auto tags = TracingLibrary::default_tags();
   if (!tags.empty()) {
-    for (const auto [key, value] : tags) {
+    for (const auto &[key, value] : tags) {
       auto ngx_value = to_ngx_str(cf->pool, value);
       auto *complex_value = datadog::common::make_complex_value(cf, ngx_value);
       if (complex_value == nullptr) {

